jetstream_pt/ray_engine_master.py

```diff
@@ -0,0 +1,176 @@
+from typing import Any, Optional, Union
+
+import numpy as np
+import jax
+import ray
+from ray.util.accelerators import tpu
+
+from jetstream.engine import engine_api, tokenizer_pb2
+from jetstream_pt.ray_engine_worker import PyTorchEngineRayWorker
+
+Params = Any
+Prefix = Any
+DecodeState = Any
+
+
+class PyTorchEngineRayMaster(engine_api.Engine):
+  """Ray engine master to orchestrate requests and collect token response"""
+
+  def __init__(
+      self, engine_workers, tokenizer_path, context_length, batch_size
+  ):
+    self.engine_workers = engine_workers
+    self.tokenizer_path = tokenizer_path
+    self.context_length = context_length
+    self.batch_size = batch_size
+
+  # pylint: disable-next=all
+  def load_params(self) -> Params:
+    all_outputs = []
+    for worker in self.engine_workers:
+      output = worker.load_params_ray.remote()
+      all_outputs.append(output)
+    _ = ray.get(all_outputs)
+    return None
+
+  # pylint: disable-next=all
+  def init_decode_state(
+      self,
+  ) -> DecodeState:
+    all_outputs = []
+    for worker in self.engine_workers:
+      output = worker.init_decode_state_ray.remote()
+      all_outputs.append(output)
+    _ = ray.get(all_outputs)
+    return None
```

Comment:
instead of returning None, is it meaningful to return some sort of "handle" here?

So basically the reason we don't need to return is because the worker will hold on the state in a local variable. However, we can also have the worker to store the state in a list and return it's index to master

master then keep a list of ints as the "handle". 

This way on good thing is that you dont need to worry about overwrite if prefill / insert is called multiple times . and helps with disagg setup

Reply:
Good point! It would be very important for disaggregation serving.  The API might have large change with disaggregation serving, can we hold it for now? We will have a clear API when we do disaggregation serving.